symengine/tests/basic/test_functions.cpp

```diff
@@ -4081,3 +4082,95 @@ TEST_CASE("test_dummy", "[Dummy]")
     CHECK(neq(*xdummy1, *x1));
     CHECK(neq(*xdummy1, *x1->as_dummy()));
 }
+
+TEST_CASE("test_sign", "[Sign]")
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const Symbol> y = symbol("y");
+    RCP<const Basic> r = sign(one);
+    RCP<const Basic> s;
+    CHECK(eq(*r, *one));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(zero);
+    CHECK(eq(*r, *zero));
+
+    r = sign(minus_one);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(real_double(1.2));
+    CHECK(eq(*r, *one));
+
+    r = sign(real_double(0.0));
+    CHECK(eq(*r, *zero));
+
+    r = sign(real_double(-1.2));
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(rational(-1, 2));
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(rational(1, 2));
+    CHECK(eq(*r, *one));
+
+    r = sign(rational(0, 2));
+    CHECK(eq(*r, *zero));
+
+    r = sign(Inf);
+    CHECK(eq(*r, *one));
+
+    r = sign(NegInf);
+    CHECK(eq(*r, *minus_one));
+
+    r = sign(ComplexInf);
+    CHECK(r->__str__() == "sign(zoo)");
+
+    r = sign(Nan);
+    CHECK(eq(*r, *Nan));
+
+    r = sign(complex_double(std::complex<double>(0, 3)));
+    CHECK(eq(*r, *I));
+
+    r = sign(complex_double(std::complex<double>(0, -3)));
+    CHECK(eq(*r, *mul(I, minus_one)));
+
+    r = sign(pi);
+    CHECK(eq(*r, *one));
+
+    r = sign(E);
+    CHECK(eq(*r, *one));
+
+    r = sign(sign(x));
+    CHECK(eq(*r, *sign(x)));
+
+    r = sign(pow(I, rational(1, 2)));
+    CHECK(r->__str__() == "sign(sqrt(I))");
+
+    r = mul(mul(integer(2), x), pow(y, integer(3)));
+    s = sign(mul(x, pow(y, integer(3))));
+    CHECK(eq(*sign(r), *s));
+    CHECK(s->__str__() == "sign(x*y**3)");
+
+    r = mul(mul(mul(integer(2), x), pow(y, integer(3))), I);
+    s = mul(sign(mul(x, pow(y, integer(3)))), I);
+    CHECK(eq(*sign(r), *s));
+    CHECK(s->__str__() == "I*sign(x*y**3)");
+
+    r = mul(mul(mul(integer(2), x), pow(y, integer(3))),
+            pow(mul(integer(3), I), integer(3)));
+    s = mul(sign(mul(x, pow(y, integer(3)))), mul(I, minus_one));
+    CHECK(eq(*sign(r), *s));
+    CHECK(s->__str__() == "-I*sign(x*y**3)");
+
+    r = sign(mul(mul(pow(Complex::from_two_nums(*integer(2), *integer(3)),
+                         Rational::from_two_ints(3, 2)),
+                     x),
+                 pow(mul(integer(3), I), integer(3))));
+    s = mul(mul(I, minus_one),
+            sign(mul(x, pow(Complex::from_two_nums(*integer(2), *integer(3)),
+                            Rational::from_two_ints(3, 2)))));
+    CHECK(eq(*r, *s));
+    CHECK(r->__str__() == "-I*sign(x*(2 + 3*I)**(3/2))");
```

Comment:
Printing tests should go to the test printing file and for this function, one test case is enough